contracts/transfer/src/transfer/host.rs

```diff
@@ -41,6 +41,18 @@ impl TransferContract {
         let generator_note =
             Note::transparent(&mut rng, generator, generator_value);
 
+        // Here we are adding the notes to the state without passing from the
+        // WebAssembly side. So we need to canonicalize them since any inner
+        // [`JubJubExtended`] would result a different form otherwise.
+        let dusk_note = Note::from_bytes(&dusk_note.to_bytes()).expect(
```

Comment:
This looks like a test case instead of production code. We are serializing/deserializing here

Reply:
This is actually the current fix agreed for the nondeterminism of JubJubExtended for the time being.